### PR TITLE
Remove Skip.if from CustomSearch tests

### PR DESCRIPTION
**Pull request by feordin**
## Description
Remove skip-if from custom search tests as they were throwing exceptions

## Related issues
Addresses [issue #].

## Testing
Describe how this change was tested.

## FHIR Team Checklist
- [ ] **Update the title** of the PR to be succinct and less than 50 characters
- [ ] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [ ] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [ ] Tag the PR with **Azure API for FHIR** if this will release to the managed service
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs [hunk 1]
```diff
@@ -72,7 +72,7 @@ public async Task GivenANewSearchParam_WhenReindexingComplete_ThenResourcesSearc
             }
             catch (FhirException ex) when (ex.StatusCode == HttpStatusCode.BadRequest && ex.Message.Contains("not enabled"))
             {
-                Skip.If(!_fixture.IsUsingInProcTestServer, "Reindex is not enabled on this server.");
+                // This test case won't run if reindex is not enabled
```
Comment (LTA-Thinking) [inline test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs:75]: Is the try catch block needed if this won't be hit?

**Comment by Ivanidzo4ka** [comment]
Is it to unblock s55 release because tests are failing?
I just not sure why it's going to that branch if this is just tests...

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs (hunk 1 above)
Comment (brendankowitz) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs:75]: The attribute needs to be `[SkippableFact]` for Skip.If to work

**Comment by feordin** [comment]
> Is it to unblock s55 release because tests are failing?
> I just not sure why it's going to that branch if this is just tests...

@Ivanidzo4ka that is correct Ivan.  Tests are failing in the S55 release, and this is to unblock the release.

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs (hunk 1 above)
Comment (feordin) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/CustomSearchParamTests.cs:75]: Updated to use the SkippableFact

**Review by LTA-Thinking — APPROVED** [review]

**Comment by brendankowitz** [comment]
Merging now, tests will run again to build the hotfix

**Pull request merged**